.circleci/config.yml

```diff
@@ -0,0 +1,162 @@
+  Sonar Code Quality Scan: # Quality-Code Scanning -- Call out what flavor (Sonar presently) and indicate Code Quality Scan
+    docker: *image
+    steps:
+    # PR Analysis cannot have branch.target parameter
+    # PR Analysis requires "pullrequest.provider" & "pullrequest.github.reposotiry" to fire decorators back to GitHub
+    # CircleCI limit:  no organization name for easier pointing to github.repository value
+      - run:
+          name: Determining Sonar and Branch Configuration
+          command: |
+              SONAR_PARAMS='';
+              if [[ ! -z $CIRCLE_PULL_REQUEST ]];
+              then
+                SONAR_PARAMS="${SONAR_PARAMS} \
+                -Dsonar.pullrequest.key=${CIRCLE_PULL_REQUEST##*/} \
+                -Dsonar.pullrequest.base=master \
+                -Dsonar.pullrequest.branch=${CIRCLE_BRANCH} \
+                -Dsonar.pullrequest.provider=GitHub
+                -Dsonar.pullrequest.github.repository=connexta/${CIRCLE_PROJECT_REPONAME}"
+              fi
+              if [[ ${CIRCLE_BRANCH} != 'master' && -z $CIRCLE_PULL_REQUEST ]];
+              then
+                SONAR_PARAMS="${SONAR_PARAMS} \
+                -Dsonar.branch.target=master \
+                -Dsonar.branch.name=${CIRCLE_BRANCH}"
+              fi
+              echo "export SONAR_PARAMS='${SONAR_PARAMS}'" >> $BASH_ENV
+              source $BASH_ENV
+      - checkout
+        # sonar.pullrequest.base should be your default branch but may not be "master"
+        # sonar.login is the API token we can use and here is configured for a CircleCI Env Variable
+        # sonar.organization "cx" is the Connexta Organization
+        # CircleCI Env. Variable CIRCLE_PULL_REQUEST only works for "forked" repos
+      - run:
+          name: Scanning
+          command: |
+            echo $SONAR_PARAMS;
+            mvn verify sonar:sonar \
+            -Dsonar.projectKey=${CIRCLE_PROJECT_REPONAME} \
+            -Dsonar.organization=cx \
+            -Dsonar.host.url=https://sonarcloud.io \
+            -Dsonar.login=$SONAR_TOKEN \
+            $SONAR_PARAMS;
+  Contract Testing: # Testing-Contract
+    docker:
+      - image: circleci/ruby:2.4.1
+    steps:
+      - checkout
+      - run: echo "Running Contract Tests"
+  Integration Testing: # Testing-Integration
+    docker:
+      - image: circleci/ruby:2.4.1
+    steps:
+      - checkout
+      - run: echo "Running Integration Tests"
+  Deploy Artifacts: # Deploy artifacts to Nexus or Maven Central
+    docker:
+      - image: circleci/ruby:2.4.1
+    steps:
+      - checkout
+      - run: echo "Deploying artifacts"
+
+# There should be a minimum of 2 workflows:
+# 1) a PR workflow for vetting through the CICD pipeline PR changes
+# 2) a master workflow for vetting the overall changes out to the public via CICD pipeline
+workflows:
+  version: 2.1 #Version indicates which CircleCI language version is in use
+  Ion Pipeline: # Defines a workflow "PR Workflow" to show up in UI
```

Comment:
@mdang8 - Could you revise the comments here and above `workflows:`, since this workflow is being used for pull requests and for master?

Reply:
This comment is effectively OBE in the proposed v1.0 of the template.
https://github.com/connexta/ion-cicd/blob/template_v1.0-rc/circleci/templates/maven/config_template-no-docker-v1.yml